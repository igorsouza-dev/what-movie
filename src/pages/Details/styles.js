import styled from 'styled-components';
import { screenSize } from 'styles/variables';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 10px;
`;

export const Backdrop = styled.div`
  background: url(${(props) => props.background});
  width: 100%;
  background-size: cover;
  background-repeat: no-repeat;
  border-radius: 4px 4px 0 0;
  display: flex;
  flex-direction: column;
`;
export const ButtonsContainer = styled.div`
  width: 100%;
  padding: 10px;
  background: rgba(255,255,255,0.6);
  display: flex;
  justify-content: center;
  align-items: center;
  > div {
    max-width: 300px;

    button {
      border-radius: 4px;
      margin: 2px;
    }
  }
  @media only screen and (min-width: 768px) {
    justify-content: flex-end;
  }
`;
export const DetailsContainer = styled.div`
  display: flex;
  flex-direction: row;
  background: rgba(255, 255, 255, 0.6);
  padding: 10px;
  position: relative;
  height: 100%;
  width: 100%;
`;
export const Score = styled.div`
  background: rgba(0, 0, 0, 0.8);
  color: #fff;
  padding: 10px;
  font-size: 12px;
  font-weight: bold;
  position: absolute;
  top: 10px;
  right: 10px;
  border-radius: 6px;
`;
export const TitleContainer = styled.div`
  margin-left: 10px;
`;
export const InfoContainer = styled.div`
  background: #fff;
  padding: 10px;
  width: 100%;
  border-radius: 0 0 4px 4px;
`;
export const Poster = styled.img`
  height: 185px;
  background: #ddd;
  border-radius: 4px;
  border: 0;
  outline: 0;
`;
export const GenreContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin: 10px;
  flex-wrap: wrap;
`;
export const Pill = styled.div`
  border-radius: 4px;
  background: #363636;
  color: #eee;
  font-size: 12px;
  width: max-content;
  padding: 2px;
  margin: 2px;
`;
export const Title = styled.h2`
  max-width: 90%;
`;
export const Subtitle = styled.h3`
  margin-top: 10px;
`;

export const VideosContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Video = styled.iframe`
  margin: 10px;
  width: 100%;
  @media only screen and (max-width: ${screenSize.mobileS}) {
    width: 250px;
    height: 120px;
  }
  @media only screen and (min-width: 321px) and (max-width: ${screenSize.mobileM}) {
    width: 300px;
    height: 140px;
  }
  @media only screen and (min-width: 376px) and (max-width: ${screenSize.mobileL}) {
    width: 336px;
    height: 157px;
  }
  @media only screen and (min-width: 426px) and (max-width: 767px)  {
    width: 386px;
    height: 181px;
  }
  @media only screen and (min-width: 768px) and (max-width: ${screenSize.laptop})  {
    width: 728px;
    height: 341px;
  }
  @media only screen and (min-width: 1025px) {
    width: 980px;
    height: 460px;
  }
`;

export const Description = styled.p``;
