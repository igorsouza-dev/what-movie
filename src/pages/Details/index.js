import React, { useEffect, useState } from 'react';
import PageContainer from 'components/PageContainer';
import { useParams } from 'react-router-dom';
import api from 'services/api';

import { Container } from './styles';

export default function Details() {
  const { id } = useParams();
  const [movie, setMovie] = useState();
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    async function getMovie() {
      try {
        const response = await api.get(
          `/movie/${id}?append_to_response=videos`
        );
        console.log(response.data);
        setMovie(response.data);
      } catch (e) {}
    }
    getMovie();
    setLoading(false);
  }, []);
  if (loading) return <div>Loading...</div>;
  return (
    <PageContainer>
      <Container
        background={
          movie &&
          `${process.env.REACT_APP_TMDB_IMAGE_URL}/original${movie.backdrop_path}`
        }
      >
        {movie && (
          <>
            <h1>{movie.original_title}</h1>
          </>
        )}
      </Container>
    </PageContainer>
  );
}
