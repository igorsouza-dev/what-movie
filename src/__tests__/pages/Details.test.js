import React from 'react';
import { cleanup, wait } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Details from 'pages/Details';

import apiMock from 'services/api';

import { renderWithRedux } from '../utils/redux';

jest.mock('services/api');

describe('<Details />', () => {
  let movie = {};
  beforeEach(() => {
    movie = {
      adult: false,
      backdrop_path: '/z2QUexmccqrvw1kDMw3R8TxAh5E.jpg',
      belongs_to_collection: null,
      budget: 0,
      genres: [{ id: 18, name: 'Drama' }, { id: 80, name: 'Crime' }, { id: 35, name: 'Comedy' }],
      homepage: null,
      id: 2,
      imdb_id: 'tt0094675',
      original_language: 'fi',
      original_title: 'Ariel',
      overview: "Taisto Kasurinen is a Finnish coal miner whose father has just committed suicide and who is framed for a crime he did not commit. In jail, he starts to dream about leaving the country and starting a new life. He escapes from prison but things don't go as planned...",
      popularity: 9.225,
      poster_path: '/gZCJZOn4l0Zj5hAxsMbxoS6CL0u.jpg',
      production_companies: [{
        id: 2303, logo_path: null, name: 'Villealfa Filmproductions', origin_country: 'FI'
      }],
      production_countries: [{ iso_3166_1: 'FI', name: 'Finland' }],
      release_date: '1988-10-21',
      revenue: 0,
      runtime: 73,
      spoken_languages: [{ iso_639_1: 'de', name: 'Deutsch' }, { iso_639_1: 'fi', name: 'suomi' }],
      status: 'Released',
      tagline: '',
      title: 'Ariel',
      video: false,
      vote_average: 7.0,
      vote_count: 90,
      videos: { results: [] }
    };
  });
  afterEach(cleanup);

  it('should render', async () => {
    const id = 2;

    apiMock.get.mockResolvedValueOnce({
      data: movie
    });
    const history = createMemoryHistory();
    history.push(`/details/${id}`);
    const { container, getByText } = renderWithRedux(
      <MemoryRouter initialEntries={[`/details/${id}`]} initialIndex={0}>
        <Route path="/details/:id">
          <Details />
        </Route>
      </MemoryRouter>
    );
    expect(apiMock.get).toBeCalledWith(`/movie/${id}?append_to_response=videos`);


    await wait(() => getByText(/Overview/i));

    expect(getByText('Ariel (1988)')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the videos section when videos are provided', async () => {
    const { id } = movie;
    movie.videos.results = [
      { key: 'ayoutubekey', site: 'YouTube' }
    ];

    apiMock.get.mockResolvedValueOnce({
      data: movie
    });
    const history = createMemoryHistory();
    history.push(`/details/${id}`);
    const { container, getByText, queryByText } = renderWithRedux(
      <MemoryRouter initialEntries={[`/details/${id}`]} initialIndex={0}>
        <Route path="/details/:id">
          <Details />
        </Route>
      </MemoryRouter>
    );


    await wait(() => getByText(/Trailers/i));

    expect(queryByText(/No video found./i)).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});
