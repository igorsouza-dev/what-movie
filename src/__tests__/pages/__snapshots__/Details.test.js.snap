// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> should render 1`] = `
<div
  class="sc-iwsKbI gWyLDK"
>
  <div
    class="sc-htoDjs dJCVbN"
  >
    <a
      class="sc-dnqmqq lcgmO"
      href="/"
    >
      <h1
        class="sc-bxivhb gZfZeg"
        color="#fff"
        data-testid="logo"
        size="40"
      >
        What Movie?
      </h1>
    </a>
    <form
      class="sc-bdVaJa hjizjZ"
      data-testid="search-bar"
    >
      <input
        class="sc-bwzfXH aYadF"
        placeholder="Type the name of a movie"
        type="text"
        value=""
      />
      <button
        class="sc-htpNat jisJQc"
        type="submit"
      >
        <svg
          color="#fff"
          fill="currentColor"
          height="24"
          size="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 512 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </button>
    </form>
    <button
      class="sc-ifAKCX gTwPhi"
      data-testid="watch-later"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
        />
      </svg>
      <div
        class="sc-EHOje sBAZD"
        data-testid="count-value"
      >
        0
      </div>
    </button>
    <button
      class="sc-bZQynM kaqpnu"
      data-testid="favorites-btn"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 576 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
        />
      </svg>
      <div
        class="sc-gzVnrw ibxOwE"
        data-testid="count-value"
      >
        0
      </div>
    </button>
  </div>
  <div
    class="sc-gqjmRU jfNrNN"
  >
    <h3>
      Favorites
    </h3>
    <ul>
      No movies added.
    </ul>
  </div>
  <div
    class="sc-gZMcBi fAhRbv"
  >
    <div
      class="sc-kAzzGY dZkYUI"
    >
      <div
        class="sc-chPdSV dpVMpT"
      >
        <div
          class="sc-kGXeez dkrMkf"
        >
          <img
            class="sc-jKJlTe PVCQD"
            src="https://image.tmdb.org/t/p/w185/gZCJZOn4l0Zj5hAxsMbxoS6CL0u.jpg"
          />
          <div
            class="sc-kpOJdX kNQtub"
          >
            7/10
          </div>
          <div
            class="sc-dxgOiQ gnjCAt"
          >
            <h2
              class="sc-kEYyzF kjjCcb"
            >
              Ariel (1988)
               
            </h2>
            <div
              class="sc-eNQAEJ gwCxre"
            >
              <div
                class="sc-hMqMXs eNHhuq"
              >
                73 min
              </div>
               - 
              <div
                class="sc-hMqMXs eNHhuq"
              >
                Drama
              </div>
              <div
                class="sc-hMqMXs eNHhuq"
              >
                Crime
              </div>
              <div
                class="sc-hMqMXs eNHhuq"
              >
                Comedy
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-kgoBCf hQuyOM"
        >
          <div
            class="sc-jTzLTM iFTcjh"
          >
            <button
              class="sc-fjdhpX sc-jzJRlG ilSPLc"
              type="button"
            >
              <svg
                color="#fff"
                fill="currentColor"
                height="20"
                size="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 448 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
              <span>
                Watch Later
              </span>
            </button>
            <button
              class="sc-fjdhpX sc-cSHVUG iAJhyH"
              data-testid="favorite"
              type="button"
            >
              <svg
                color="#fff"
                fill="currentColor"
                height="24"
                size="24"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 576 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-ckVGcZ eztdpj"
      >
        <h3
          class="sc-kkGfuU kriFMi"
        >
          Overview
        </h3>
        <p
          class="sc-eHgmQL gmKbQu"
        >
          Taisto Kasurinen is a Finnish coal miner whose father has just committed suicide and who is framed for a crime he did not commit. In jail, he starts to dream about leaving the country and starting a new life. He escapes from prison but things don't go as planned...
        </p>
        <h3
          class="sc-kkGfuU kriFMi"
        >
          Trailers
        </h3>
        <div
          class="sc-iAyFgw jrTbEH"
        >
          <p
            class="sc-eHgmQL gmKbQu"
          >
            No video found.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div>
    Film data provided by TMDB
  </div>
</div>
`;

exports[`<Details /> should render the videos section when videos are provided 1`] = `
<div
  class="sc-iwsKbI gWyLDK"
>
  <div
    class="sc-htoDjs dJCVbN"
  >
    <a
      class="sc-dnqmqq lcgmO"
      href="/"
    >
      <h1
        class="sc-bxivhb gZfZeg"
        color="#fff"
        data-testid="logo"
        size="40"
      >
        What Movie?
      </h1>
    </a>
    <form
      class="sc-bdVaJa hjizjZ"
      data-testid="search-bar"
    >
      <input
        class="sc-bwzfXH aYadF"
        placeholder="Type the name of a movie"
        type="text"
        value=""
      />
      <button
        class="sc-htpNat jisJQc"
        type="submit"
      >
        <svg
          color="#fff"
          fill="currentColor"
          height="24"
          size="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 512 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </button>
    </form>
    <button
      class="sc-ifAKCX gTwPhi"
      data-testid="watch-later"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
        />
      </svg>
      <div
        class="sc-EHOje sBAZD"
        data-testid="count-value"
      >
        0
      </div>
    </button>
    <button
      class="sc-bZQynM kaqpnu"
      data-testid="favorites-btn"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 576 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
        />
      </svg>
      <div
        class="sc-gzVnrw ibxOwE"
        data-testid="count-value"
      >
        0
      </div>
    </button>
  </div>
  <div
    class="sc-gqjmRU jfNrNN"
  >
    <h3>
      Favorites
    </h3>
    <ul>
      No movies added.
    </ul>
  </div>
  <div
    class="sc-gZMcBi fAhRbv"
  >
    <div
      class="sc-kAzzGY dZkYUI"
    >
      <div
        class="sc-chPdSV dpVMpT"
      >
        <div
          class="sc-kGXeez dkrMkf"
        >
          <img
            class="sc-jKJlTe PVCQD"
            src="https://image.tmdb.org/t/p/w185/gZCJZOn4l0Zj5hAxsMbxoS6CL0u.jpg"
          />
          <div
            class="sc-kpOJdX kNQtub"
          >
            7/10
          </div>
          <div
            class="sc-dxgOiQ gnjCAt"
          >
            <h2
              class="sc-kEYyzF kjjCcb"
            >
              Ariel (1988)
               
            </h2>
            <div
              class="sc-eNQAEJ gwCxre"
            >
              <div
                class="sc-hMqMXs eNHhuq"
              >
                73 min
              </div>
               - 
              <div
                class="sc-hMqMXs eNHhuq"
              >
                Drama
              </div>
              <div
                class="sc-hMqMXs eNHhuq"
              >
                Crime
              </div>
              <div
                class="sc-hMqMXs eNHhuq"
              >
                Comedy
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-kgoBCf hQuyOM"
        >
          <div
            class="sc-jTzLTM iFTcjh"
          >
            <button
              class="sc-fjdhpX sc-jzJRlG ilSPLc"
              type="button"
            >
              <svg
                color="#fff"
                fill="currentColor"
                height="20"
                size="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 448 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
              <span>
                Watch Later
              </span>
            </button>
            <button
              class="sc-fjdhpX sc-cSHVUG iAJhyH"
              data-testid="favorite"
              type="button"
            >
              <svg
                color="#fff"
                fill="currentColor"
                height="24"
                size="24"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 576 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-ckVGcZ eztdpj"
      >
        <h3
          class="sc-kkGfuU kriFMi"
        >
          Overview
        </h3>
        <p
          class="sc-eHgmQL gmKbQu"
        >
          Taisto Kasurinen is a Finnish coal miner whose father has just committed suicide and who is framed for a crime he did not commit. In jail, he starts to dream about leaving the country and starting a new life. He escapes from prison but things don't go as planned...
        </p>
        <h3
          class="sc-kkGfuU kriFMi"
        >
          Trailers
        </h3>
        <div
          class="sc-iAyFgw jrTbEH"
        >
          <iframe
            allowfullscreen=""
            class="sc-hSdWYo hcZNGB youtube-player"
            frameborder="0"
            src="https://www.youtube.com/embed/ayoutubekey?autoplay=0"
            title="YouTube"
            type="text/html"
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    Film data provided by TMDB
  </div>
</div>
`;
